pil-analyzer/src/pil_analyzer.rs

```diff
@@ -428,14 +428,7 @@ impl PILAnalyzer {
                     .unwrap(),
             )
             .unwrap();
-            if let Some(degree) = self.polynomial_degree {
-                assert_eq!(
-                    degree, namespace_degree,
-                    "all namespaces must have the same degree"
-                );
-            } else {
-                self.polynomial_degree = Some(namespace_degree);
-            }
+            self.polynomial_degree = Some(namespace_degree);
         }
```

Comment:
I would set `polynomial_degree` to `None` here. Otherwise, the next namespace "inherits" the previous namespace's degree if it's not set.